src/main/java/org/spongepowered/api/command/CommandLowLevel.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command;
+
+import org.spongepowered.api.command.specification.CommandSpec;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * A low-level interface for commands that can be executed. For almost all use
+ * cases, higher-level tools should be used instead, like
+ * {@link CommandSpec}.
+ *
+ * <p>Implementations are not required to implement a sane
+ * {@link Object#equals(Object)} but really should.</p>
+ */
+public interface CommandLowLevel {
```

Comment:
I'll keep that in mind. I also plan to write extensive documentation for SpongeDocs, based on the concepts I write in comments here, as well as a migration guide.

My fear is that pulling this and breaking everything is going to put off developers, giving an API version or two buffer will give some grace and allow developers and users to transition in a more comfortable timeline. Breaking almost all plugins with a single swipe without warning is not good for the ecosystem, no matter how good it might be for us as Sponge developers, I do not think it will be good for the community or those pluming developers that do not have a lot of time to spend on their plugins.

If it is decided to break things outright, then that will be easy to do at any point. Having to put backwards compat back in is much harder - this is my actual reason for keeping backwards compat for now, if it is decided that it should stay, rather than go. It makes naming things harder, yes, and it is a challenge to make sure the old system mostly conforms to the new - but I'd rather solve that challenge now than be told I have to face that challenge much later on.